### ISSUE TITLE
Improving accessibility support of Recharts - Contribution offer

### ISSUE THREAD

**Issue by cscrosati** [recharts/recharts#2801]
- [x ] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Developers using Recharts today don't have a comprehensive set of features and guidance on creating accessible charts for their end customers.

Developers are currently asking for this via the [Issues](https://github.com/recharts/recharts/issues?q=is%3Aissue accessibility) and [Discussions], and a couple of [Discussions](https://github.com/recharts/recharts/discussions) threads. With legal requirements to meet accessibility standards continually expanding, developers need to be able to create accessible charts with Recharts, or they will use a different library.

The end customers benefits from improved accessibility support in many aspects, including but not limited to:
- Improved contrast between data sets, which improved both low vision users. It also greatly improves visibility of data when used with low contrast projectors.
- Improved keyboard only navigation of the data shown in the chart. This benefits users with reduced motor skills, along with users who prefers keyboard only usage.
- Improved screen reader user experience in understanding the graphically displayed data with properly labeled roles and components.

We have assessed the Recharts library and identified areas of improvement and would like to collaborate with the Recharts on contributing a good chunk of accessibility features. Some are APIs additions, and some are documentation additions with guidance for developers. This feature request is a starting point for a discussion on how we can help achieve this with the Recharts maintainers.

Our ultimate goal would be to enable developers using Recharts to meet [WCAG 2.2](https://www.w3.org/TR/WCAG22/) "AA" level compliance requirements.

### What does the proposed API look like?
There is no finite API for this feature request because it encompasses a lot of details related to different aspects of accessibility. We recognize that not all of this will be API fixable, and will be instead guidance within the Recharts documentation.

Here are a few identified things that could be done through API, or through documentation:
- Expose a way to consume WAI-ARIA attributes on the charts.
- Document how to add proper roles and labels to component of a chart.
- Associate good default roles for major components of Rechart.
- Include a default ARIA role for all charts (currently none, proposal could be FIGURE or GROUP or IMG, depending on certain criterias)
- Improve Tooltip to behave in accordance to the "[Content on Hover](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html)" principles
- Offer a default or optional theme that meets color contrast requirements for non-text elements, preserve current contrast for text elements are these meet the requirements today.

This list is not complete.

We want to reiterate we are interested in contributing a lot of this, both for code and documentation.

**Comment by fyoung-cpi** [comment]
Any work on this would be a great addition to Recharts.

**Comment by cscrosati** [comment]
I published the simplest PR change to enable some basic scenarios in order to gauge the velocity this can be iterated upon.

We have more involved changes to suggest, along with future PRs to support them. We just need a signal from the maintainer that a conversation can be had on how best to enable accessible charts with an API that matches the project philosophy.

**Comment by brendanmoore** [comment]
I am glad I found this issue and I am keen to help contribute.

> Include a default ARIA role for all charts (currently none, proposal could be FIGURE or GROUP or IMG, depending on certain criterias)

I am looking into how to go a little deeper with respect specific chart roles `graphics-object`, `graphics-symbol` etc (https://www.w3.org/wiki/SVG_Accessibility/ARIA_roles_for_charts)

**Comment by cscrosati** [comment]
@brendanmoore the aria role for charts is a proposal from W3C that doesn't seem to ever have been completed. I have not found much support in web browsers. It appears most charting library usually adopt the ARIA roles.

**Comment by mgifford** [comment]
I would point folks to https://blog.tenon.io/accessible-charts-with-aria/ & https://tink.uk/accessible-svg-line-graphs/ when looking for examples of how to build accessible SVG graphs.

**Comment by nikolasrieble** [comment]
Dear @cscrosati, this is a great idea! The best contribution possible right now would be a gap analysis, in the form of issues that are actionable. We could also consider to start measuring our accessibility in storybook and integrate an addon for that manner.

**Comment by sjsakib** [comment]
Hello, currently I have  a small [pet project](https://gold-price.sakib.dev/) with recharts where only a simple line chart is shown. When the points are hovered, some more details are shown in the tooltip. I would have loved the line to be navigable  with arrow keys.

I want to contribute that part myself here. So the idea is:

* Make the line or chart to be focusable with tab key, this activates the first point of the line and shows the tooltip.
* Move the active point with arrow keys.

Primarily I want to do it for the line chart. But maybe other type of charts can benefit form this as well.

What do you guys think about the whole idea?

**Comment by ckifer** [comment]
Hi @sjsakib 

This is already possible. Add the prop accessibilityLayer to your LineChart and you should be good to go. This is a new-ish feature and it's documented in the storybook

**Comment by mariusz97m** [comment]
> Hi @sjsakib
> 
> This is already possible. Add the prop accessibilityLayer to your LineChart and you should be good to go. This is a new-ish feature and it's documented in the storybook

Do it work properly with enabled Windows Screen Reader (Narrator) or macOS VoiceOver?
With disabled Scren Reader keyboard navigation works properly for me in example with accessibilityLayer. When I turn on my screen reader then navigation with keyboard stop works:

https://release--63da8268a0da9970db6992aa.chromatic.com/?path=/docs/api-accessibility--docs

**Comment by ckifer** [comment]
Read over this thread https://github.com/recharts/recharts/discussions/4662#discussioncomment-9770241

It answers why it stops working when voiceover is enabled

**Comment by abel-eero** [comment]
What is the current status of this?

**Comment by ckifer** [comment]
accessibilityLayer adds support for all categorical charts (Bar, Line, Area, Scatter). Polar charts and some of the others don't quite have it just yet in 2.x but they got added in 3.x but need tested. 

Other than that there's been a few minor improvements and fixes based on issues people have created. 

No huge overhauls or anything
